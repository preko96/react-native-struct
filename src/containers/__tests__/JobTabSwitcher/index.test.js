import React from 'react'
import { JobsTabSwitcher } from '../../JobsTabSwitcher/index'
import Details from '../../JobsTabSwitcher/details'
import FormList from '../../JobsTabSwitcher/formList'
import { Button } from 'native-base'

jest.mock('react-native-sqlite-storage', () => {
	const mockSQLite = {
		openDatabase: () => {
			return {
				transaction: () => {
					() => { return [] }
				}
			}
		}
	}
	return mockSQLite
})

it('renders without crashing', () => {
	expect(shallow(<JobsTabSwitcher />)).toMatchSnapshot()
})

it('expect init activeTab to be details', () => {
	const activeTab = shallow(<JobsTabSwitcher />).state('activeTab')
	expect(activeTab).toEqual('details')
})

it('onPressTab works properly', () => {
	const component = shallow(<JobsTabSwitcher />).instance()
	component.onPressTab('test')
	const activeTabAfterPress = component.state.activeTab
	expect(activeTabAfterPress).toEqual('test')
})

it('renders details when activeTab equals details', () => {
	const wrapper = shallow(<JobsTabSwitcher />)
	wrapper.setState({ activeTab: 'details' })
	const details = wrapper.find(Details)
	expect(details).toHaveLength(1)
})

it('renders formsList when activeTab equals forms', () => {
	const wrapper = shallow(<JobsTabSwitcher />)
	wrapper.setState({ activeTab: 'forms' })
	const details = wrapper.find(FormList)
	expect(details).toHaveLength(1)
})

it('set activeTab state to details when details button is pressed', () => {
	const wrapper = shallow(<JobsTabSwitcher />)
	//change activeTab to test the behaviour
	wrapper.setState({ activeTab: 'not details' })

	const detailsButton = wrapper.find(Button).first()
	detailsButton.simulate('Press')
	const activeTab = wrapper.state('activeTab')
	expect(activeTab).toEqual('details')
})

it('set activeTab state to forms when forms button is pressed', () => {
	const wrapper = shallow(<JobsTabSwitcher />)
	//change activeTab to test the behaviour
	wrapper.setState({ activeTab: 'not forms' })
	
	const formsButton = wrapper.find(Button).last()
	formsButton.simulate('Press')
	const activeTab = wrapper.state('activeTab')
	expect(activeTab).toEqual('forms')
})

