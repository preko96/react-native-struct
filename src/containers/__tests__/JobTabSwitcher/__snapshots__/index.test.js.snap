// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Component>
  <Component>
    <Connect(RegDynamicSize)
      name="SEGMENT"
    >
      <Styled(H3)
        allowFontScaling={false}
        numberOfLines={1}
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 20,
          }
        }
      >
        undefined undefined, undefined
      </Styled(H3)>
      <Styled(Segment)
        style={
          Object {
            "height": "auto",
          }
        }
      >
        <Styled(Button)
          active={true}
          first={true}
          full={true}
          onPress={[Function]}
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Styled(Text)
            allowFontScaling={false}
          >
            Details
          </Styled(Text)>
        </Styled(Button)>
        <Styled(Button)
          active={false}
          full={true}
          last={true}
          onPress={[Function]}
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Styled(Text)
            allowFontScaling={false}
          >
            Forms
          </Styled(Text)>
        </Styled(Button)>
      </Styled(Segment)>
      <Component
        style={
          Object {
            "backgroundColor": "#e7e7e7",
            "height": 1,
            "marginBottom": 10,
            "marginTop": 10,
            "width": "100%",
          }
        }
      />
    </Connect(RegDynamicSize)>
  </Component>
  <Component
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Details) />
  </Component>
  <Component
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Connect(FormList) />
  </Component>
</Component>
`;
